test/runtests.jl

```diff
@@ -191,6 +192,39 @@ end
     @test @inferred(f9(plan_fft(zeros(10), 1), 10)) == 1/10
 end
 
+@testset "adjoint" begin
+    @testset "complex fft adjoint" begin
+        for x in (randn(3), randn(3, 4), randn(3, 4, 5))
+            N = ndims(x)
+            y = randn(size(x))
+            for dims in unique((1, 1:N, N))
+                P = plan_fft(x, dims)
+                @test dot(y, P * x) ≈ dot(P' * y, x)
+                @test_broken dot(y, P \ x) ≈ dot(P' \ y, x)
```

Comment:
While the inverse of an adjoint plan is not needed for the rrule's, I tried to include it for completeness. However, I was getting an error (which only occurs with the test plans seemingly, not fftw) that I just could not fix.

```
TypeError: in typeassert, expected AbstractFFTs.ScaledPlan{ComplexF64, TestPlan{ComplexF64, 3, UnitRange{Int64}}, ComplexF64}, got a value of type TestPlan{ComplexF64, 3, UnitRange{Int64}}
  Stacktrace:
    [1] inv(p::InverseTestPlan{ComplexF64, 3, UnitRange{Int64}})
      @ AbstractFFTs ~/Desktop/repos/AbstractFFTs.jl/src/definitions.jl:226
    [2] \(p::InverseTestPlan{ComplexF64, 3, UnitRange{Int64}}, x::Array{Float64, 3})
      @ AbstractFFTs ~/Desktop/repos/AbstractFFTs.jl/src/definitions.jl:228
    [3] *(p::AbstractFFTs.AdjointPlan{ComplexF64, InverseTestPlan{ComplexF64, 3, UnitRange{Int64}}}, x::Array{Float64, 3})
      @ AbstractFFTs ~/Desktop/repos/AbstractFFTs.jl/src/definitions.jl:611
    [4] *
      @ ~/Desktop/repos/AbstractFFTs.jl/src/definitions.jl:251 [inlined]
    [5] \(p::AbstractFFTs.AdjointPlan{ComplexF64, TestPlan{ComplexF64, 3, UnitRange{Int64}}}, x::Array{Float64, 3})
      @ AbstractFFTs ~/Desktop/repos/AbstractFFTs.jl/src/definitions.jl:632
    [6] macro expansion
      @ /Applications/Julia-1.7.app/Contents/Resources/julia/share/julia/stdlib/v1.7/Test/src/Test.jl:445 [inlined]
    [7] macro expansion
      @ ~/Desktop/repos/AbstractFFTs.jl/test/runtests.jl:203 [inlined]
    [8] macro expansion
      @ /Applications/Julia-1.7.app/Contents/Resources/julia/share/julia/stdlib/v1.7/Test/src/Test.jl:1283 [inlined]
    [9] macro expansion
      @ ~/Desktop/repos/AbstractFFTs.jl/test/runtests.jl:197 [inlined]
   [10] macro expansion
      @ /Applications/Julia-1.7.app/Contents/Resources/julia/share/julia/stdlib/v1.7/Test/src/Test.jl:1283 [inlined]
   [11] top-level scope
      @ ~/Desktop/repos/AbstractFFTs.jl/test/runtests.jl:196
```

Reply:
Still unresolved. Minimal steps to reproduce
```
using AbstractFFTs
using AbstractFFTs: Plan
include("test/testplans.jl")
p = plan_fft(rand(3))
inv(p') * rand(3)
```

Reply:
Resolved by https://github.com/JuliaMath/AbstractFFTs.jl/pull/69